packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -51,3 +62,235 @@ export async function fetchSnapshot(
+            for (let i = resources1.length - 1; i > 0; i--) {
+                const indResTime = resources1[i] as PerformanceResourceTiming;
+                const resource_name = indResTime.name;
+                const resource_initiatortype = indResTime.initiatorType;
+                if ((resource_initiatortype.localeCompare("fetch") === 0) && (resource_name.localeCompare(url) === 0)) {
+                    redirectTime = indResTime.redirectEnd - indResTime.redirectStart;
+                    dnstime = indResTime.domainLookupEnd - indResTime.domainLookupStart;
+                    tcpHandshakeTime = indResTime.connectEnd - indResTime.connectStart;
+                    secureConntime = (indResTime.secureConnectionStart > 0) ?
+                        (indResTime.connectEnd - indResTime.secureConnectionStart) : 0;
+                    responseTime = indResTime.responseEnd - indResTime.responseStart;
+                    fetchStToRespEndTime = (indResTime.fetchStart > 0) ?
+                        (indResTime.responseEnd - indResTime.fetchStart) : 0;
+                    reqStToRespEndTime = (indResTime.requestStart > 0) ?
+                        (indResTime.responseEnd - indResTime.requestStart) : 0;
+                    networkTime = (indResTime.startTime > 0) ? (indResTime.responseEnd - indResTime.startTime) : 0;
+                    if (spReqDuration) {
+                        networkTime = networkTime - parseInt(spReqDuration, 10);
+                    }
+                    break;
+                }
+            }
+
+            const { numTrees, numBlobs, encodedBlobsSize, decodedBlobsSize } = evalBlobsAndTrees(snapshot);
+            const clientTime = networkTime ? overallTime - networkTime : undefined;
+
+            // There are some scenarios in ODSP where we cannot cache, trees/latest will explicitly tell us when we
+            // cannot cache using an HTTP response header.
+            const canCache = response.headers.get("disablebrowsercachingofusercontent") !== "true";
+            // There maybe no snapshot - TreesLatest would return just ops.
+            const sequenceNumber: number = (snapshot.trees && (snapshot.trees[0] as any).sequenceNumber) ?? 0;
+            const seqNumberFromOps = snapshot.ops && snapshot.ops.length > 0 ?
+                snapshot.ops[0].sequenceNumber - 1 :
+                undefined;
+
+            const value: ISnapshotCacheValue = { snapshot, sequenceNumber };
+            let cacheP: Promise<void> | undefined;
+            if (!Number.isInteger(sequenceNumber)
+                || seqNumberFromOps !== undefined && seqNumberFromOps !== sequenceNumber) {
+                logger.sendErrorEvent({ eventName: "fetchSnapshotError", sequenceNumber, seqNumberFromOps });
+                value.sequenceNumber = undefined;
+            } else if (canCache) {
+                if (epochTracker !== undefined) {
```

Comment:
Is there simple way to move caching logic out of this function and make it responsibility of callers?
Such that it does not need to deal with differences between presence and not presence of epochTracker?
Or alternatively, always go through persistedCache path (epochTracker object ideally should not expose cache APIs on it, we should separate concerns and have epoch object only for calls, and telemetry interface (implemented by it) be a separate thing (as we pass things through layers).

Reply:
Always using persisted cache to put the snapshot. Will remove the dep on epoch tracker now.